File: src/main/java/com/hedvig/claims/events/BackofficeClaimCreatedEvent.java
package com.hedvig.claims.events;

import com.hedvig.claims.aggregates.ClaimSource;
import lombok.Value;
import org.axonframework.serialization.Revision;

import java.time.Instant;
import java.util.UUID;

@Value
@Revision("1.0")
public class BackofficeClaimCreatedEvent {
  public String id;
  public String memberId;
  public Instant registrationDate;
  public ClaimSource claimSource;
  public UUID contractId;
}

File: src/main/java/com/hedvig/claims/events/BackofficeClaimCreatedEvent.kt
package com.hedvig.claims.events

import com.hedvig.claims.aggregates.ClaimSource
import java.time.Instant

data class BackofficeClaimCreatedEvent(
  val id: String,
  val memberId: String,
  val registrationDate: Instant,
  val claimSource: ClaimSource
)
